File: WebApplication/Models/GameState.cs
using CocaineCartels.BusinessLogic;

namespace CocaineCartels.WebApplication.Models
{
    public class GameState
    {
        public GameState(string currentPlayerColor, Turn currentTurn, Game gameInstance)
        {
            CurrentPlayerColor = currentPlayerColor;
            CurrentTurn = currentTurn;
            GameInstance = gameInstance;
        }

        public string CurrentPlayerColor { get; }
        public Turn CurrentTurn { get; }
        public Game GameInstance { get; }
    }
}

File: WebApplication/Client/GameState.ts
module CocaineCartels {
    "use strict";

    export class GameState {
        constructor(
            gameStateData: IGameState
        ) {
            this.gameInstance = new Game(gameStateData.currentTurn, gameStateData.gameInstance);
            this.currentPlayer = this.gameInstance.getPlayer(gameStateData.currentPlayerColor);
        }

        public currentPlayer: Player;
        public gameInstance: Game;
    }
}
